### DISCUSSION TITLE
GF2 sparse arrays supported?

### DISCUSSION THREAD

**Discussion by amirebrahimi** [mhostetter/galois#419] [Q&A] [answered]
From what I can tell sparse arrays are not supported. Is that correct?

**Comment by mhostetter** [comment, marked as answer]
SciPy isn't currently tested against. This morning I tried instantiating a `scipy.sparse` matrix and getting the finite field ufuncs to trigger and could not.

For instance, I found these methods `__add__()`, `_add_dense()`, and `_add_spare()` in `scipy.sparse.bsr_array`. However, I wasn't able to get any to use the custom finite field ufuncs constructed by `FieldArray` -- addition is available as `FieldArray._add.ufunc`, for example.

I would be happy to support them, but I haven't found the best integration with NumPy ufuncs / Numba. The principle mechanism that `galois` overrides of `ndarray` in `FieldArray` is `__array_ufunc__()`, which handles the ufunc dispatching. I haven't found such a hook in `scipy.sparse` matrices.

**Reply by amirebrahimi** [reply]
Okay. Well, thank you for looking into it and also for creating `galois`!
